From 463123fe0c189fb3950b3d9c76adcc9daca496c4 Mon Sep 17 00:00:00 2001
From: Shawn Xiao <b49994@freescale.com>
Date: Wed, 26 Aug 2015 16:10:45 +0800
Subject: [PATCH] MGS-955-2 [#1938]file_node struct change

The file struct has changed since 3.19. Changed the usage in GPU driver
too.

Date Sep 14, 2015

Signed-off-by: Shawn Xiao <b49994@freescale.com>
---
 drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_debugfs.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_debugfs.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_debugfs.c
index 0ce107c..ffcaf2f 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_debugfs.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/gc_hal_kernel_debugfs.c
@@ -559,7 +559,11 @@ _DebugFSRead (
     caddr_t data_to_return ;
     gcsDEBUGFS_Node* node ;
     /* get the metadata about this emlog */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
     if ( ( node = _GetNodeInfo ( file->f_dentry->d_inode ) ) == NULL )
+#else
+    if ( ( node = _GetNodeInfo ( file_inode(file) ) ) == NULL )
+#endif
     {
         printk ( "debugfs_read: record not found\n" ) ;
         return - EIO ;
@@ -626,7 +630,11 @@ _DebugFSWrite (
     gcsDEBUGFS_Node*node ;
 
     /* get the metadata about this log */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,1,0)
     if ( ( node = _GetNodeInfo ( file->f_dentry->d_inode ) ) == NULL )
+#else
+    if ( ( node = _GetNodeInfo ( file_inode(file) ) ) == NULL )
+#endif    
     {
         return - EIO ;
     }
-- 
1.8.0

